plugin.video.freeform/resources/lib/scraper.py

```diff
@@ -0,0 +1,108 @@
+                      'Studio':'Freeform'}
+          if getFileData == False:
+              ilist = self.addMenuItem(name,'GV', ilist, url, thumb, fanart, infoList, isFolder=False)
+          else:
+              ilist.append((infoList['Season'], infoList['Episode'], url))
+      return(ilist)
+
+
+  def doFunction(self, url):
+      func = url[0:2]
+      url  = url[2:]
+      if func == 'AL':
+        name  = xbmc.getInfoLabel('ListItem.Title')
+        profile = self.addon.getAddonInfo('profile')
+        moviesDir  = xbmc.translatePath(os.path.join(profile,'TV Shows'))
+        movieDir  = xbmc.translatePath(os.path.join(moviesDir, name))
+        if not os.path.isdir(movieDir):
+            os.makedirs(movieDir)
+        ilist = []
+        ilist = self.getAddonEpisodes(url, ilist, getFileData = True)
+        for season, episode, url in ilist:
+            se = ''.join(['S',str(season),'E',str(episode)])
+            xurl = ''.join([sys.argv[0],'?mode=GV&url=',url])
+            strmFile = xbmc.translatePath(os.path.join(movieDir, se+'.strm'))
+            with open(strmFile, 'w') as outfile:
+               outfile.write(xurl)         
+      json_cmd = '{"jsonrpc":"2.0","method":"VideoLibrary.Scan", "params": {"directory":"%s/"},"id":1}' % movieDir.replace('\\','/')
+      jsonRespond = xbmc.executeJSONRPC(json_cmd)
+
+
+  def getAddonVideo(self,url):
+      xurl = ''.join(['https://api.pluto.watchabc.go.com/api/ws/pluto/v1/module/videoplayer/2185739?brand=002&device=001&authlevel=1&layout=2185738&video=', str(url)])
+      html = requests.get(xurl, headers=self.defaultHeaders).text
+      ua = re.compile('"ULNK","value":"(.+?)"', re.DOTALL).search(html).group(1)
+      udata = ''.join(['video%5Fid=',str(url),'&device=001&video%5Ftype=lf&brand=002'])
+      url = 'https://api.entitlement.watchabc.go.com/vp2/ws-secure/entitlement/2020/authorize.json'
+      uheaders = self.defaultHeaders.copy()
+      uheaders.update({'Content-Type':'application/x-www-form-urlencoded',
+                       'Accept':'application/json',
+                       'X-Requested-With':'ShockwaveFlash/24.0.0.194',
+                       'Origin':'http://cdn1.edgedatg.com',
+                       'DNT':'1',
+                       'Referer':'http://cdn1.edgedatg.com/aws/apps/datg/web-player-unity/1.0.6.13/swf/player_vod.swf',
+                       'Pragma':'no-cache',
+                       'Connection':'keep-alive',
+                       'Cache-Control':'no-cache'})
+      a = requests.post(url, data=udata, headers=uheaders).json()
+      if a.get('uplynkData') is None:
+          xbmcgui.Dialog().notification(self.addonName, self.addon.getLocalizedString(30001), self.addonIcon)
+          return
+      url = ''.join([ua,'?',a['uplynkData']['sessionKey']])
+      liz = xbmcgui.ListItem(path = url, offscreen=True)
+# No need to process subtitles, all shows have closed captions
+      liz.setProperty('inputstreamaddon','inputstream.adaptive')
```

Comment:
This needs to be fixed, see: https://forum.kodi.tv/showthread.php?tid=353560

Reply:
Yeah now I remember that. Still don't think it was a good idea with such short notice though.